### PR TITLE
PUBDEV-8875: Support eval_metric on validation dataset in External XGBoost cluster

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/exec/RemoteXGBoostExecutorTest.java [hunk 1]
```diff
@@ -91,4 +93,34 @@ public void prostateRegressionWithCustomEarlyStopping() {
...
+            parms._valid = splits[1]._key;
+            parms._response_column = "AGE";
+            parms._ignored_columns = new String[]{"ID"};
+            parms._ntrees = 30;
+            parms._eval_metric = "rmse";
+            parms._score_tree_interval = 3;
+            parms._score_eval_metric_only = true;
+
+            XGBoostModel model = new XGBoost(parms).trainModel().get();
+            Scope.track_generic(model);
```
Comment (valenad1) [inline h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/exec/RemoteXGBoostExecutorTest.java:120]: Can we have some assert here? I have run it on master and it still passing, so I am not sure if it is testing anything.

Comment (michalkurka) [reply h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/exec/RemoteXGBoostExecutorTest.java:120]: Yes, very good point it is missing the check!

Comment (michalkurka) [reply h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/exec/RemoteXGBoostExecutorTest.java:122]: Fixed, thanks again!

**Review by valenad1 — APPROVED** [review]
LGTM, Thank you! 👍

**Comment by michalkurka** [comment]
Further tests were added in https://github.com/h2oai/h2o-3/pull/6392 (passes). Merging now.

**Pull request merged**